app/controllers/users/two_factor_authentication_controller.rb

```diff
@@ -177,6 +177,11 @@ def handle_telephony_result(method:, default:)
           otp_make_default_number: default,
           reauthn: reauthn?,
         )
+      elsif @telephony_result.error.is_a?(Telephony::OptOutError) &&
+            IdentityConfig.store.sms_resubscribe_enabled
+        flash.delete(:info)
+        user_session[:phone_id] = phone_configuration.id if phone_configuration&.phone
+        redirect_to login_two_factor_sms_opt_in_path
```

Comment:
If the first line is unable to assign to the session due to a `nil` `phone_configuration`, would the redirect result in a 404?

i.e.

- Drop the `&.` guarding if we're feeling confident it's there?
- Otherwise, include the redirect in the `if` block?

Reply:
There are other cases where we have `:unconfirmed_phone` in the session already and it still works

In an ideal world, I'd love to throw a `/sms/:phone_configuration_id/opt_in` in the URL to avoid this awkward session dance, but we only save confirmed phones to the DB :[